packages/e2e-tests/tests/chat-list-multiselect.spec.ts

```diff
@@ -0,0 +1,481 @@
+    await expect(getChat(2).getByLabel('Mute')).toBeVisible()
+    await expect(getChat(7).getByLabel('Mute')).toBeVisible()
+    await expectSelectedChats([7, 2])
+    await getChat(2).click({
+      button: 'right',
+    })
+    await expect(page.getByRole('menu').getByRole('menuitem')).toHaveText([
+      'Pin Chat',
+      'Unmute',
+      'Archive Chat',
+      'Delete Chat',
+    ])
+    await page.keyboard.press('Escape')
+
+    await getChat(4).click({
+      modifiers: ['ControlOrMeta'],
+    })
+    await expectSelectedChats([7, 4, 2])
+    await getChat(2).click({
+      button: 'right',
+    })
+    // Some of the selected are muted, some are not.
+    await expect(page.getByRole('menu').getByRole('menuitem')).toHaveText([
+      'Pin Chat',
+      'Mute Notifications',
+      'Archive Chat',
+      'Delete Chat',
+    ])
+    await page.keyboard.press('Escape')
+  })
+})
+
+test('resets selection when the active chat changes', async () => {
+  await getChat(7).click()
+  await expectSelectedChats([7])
+  await getChat(5).click({
+    modifiers: ['ControlOrMeta'],
+  })
+  await expectSelectedChats([7, 5])
+
+  await page.keyboard.press('ControlOrMeta+PageDown')
+  await expectSelectedChats([6])
+})
+
+test("selection doesn't reset if items get reordered", async () => {
+  await getChat(7).click()
+  await expectSelectedChats([7])
+  await getChat(5).click({
+    modifiers: ['ControlOrMeta'],
+  })
+  await expectSelectedChats([7, 5])
+
+  await getChat(5).click({
+    button: 'right',
+  })
+  // Change the order of chats by pinning some.
+  await page.getByRole('menuitem', { name: 'Pin Chat' }).click()
+  await expect(chatList.getByRole('tab', { name: 'Some chat ' })).toContainText(
+    [
+      'Some chat 7',
+      'Some chat 5',
+
+      'Some chat 9',
+      'Some chat 8',
+      'Some chat 6',
+      'Some chat 4',
+      'Some chat 3',
+      'Some chat 2',
+      'Some chat 1',
+    ]
+  )
+  await expectSelectedChats([7, 5])
+
+  await getChat(5).click({
+    button: 'right',
+  })
+  // Unpin all, to restore state for other tests.
+  await page.getByRole('menuitem', { name: 'Unpin Chat' }).click()
+  await expect(chatList.getByRole('tab', { name: 'Some chat ' })).toContainText(
+    [
+      'Some chat 9',
+      'Some chat 8',
+      'Some chat 7',
+      'Some chat 6',
+      'Some chat 5',
+      'Some chat 4',
+      'Some chat 3',
+      'Some chat 2',
+      'Some chat 1',
+    ]
+  )
+})
+
+test('delete several', async () => {
+  await getChat(7).click()
+  await expectSelectedChats([7])
+  await getChat(5).click({
+    modifiers: ['Shift'],
+  })
+  await getChat(2).click({
+    modifiers: ['ControlOrMeta'],
+  })
+  await expectSelectedChats([7, 6, 5, 2])
+
+  await getChat(6).click({
+    button: 'right',
+  })
+  await page.getByRole('menuitem', { name: 'Delete Chat' }).click()
+
+  await expect(page.getByRole('dialog')).toContainText(
+    'Delete 4 chats on all your devices?'
```

Comment:
This will fail if the translation changes, but I agrree on the decision better have a failing test after a translation update then missing if the dialog does not show the expected text